systemtests/tests/checkpoints/testrunner-checkpoints-on-cancel

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+set -e
+set -o pipefail
+set -u
+#
+# Cancel a backup and check that metadata is still saved with checkpoints.
+#
+TestName="$(basename "$(pwd)")"
+export TestName
+
+#shellcheck source=../environment.in
+. ./environment
+
+#shellcheck source=../scripts/functions
+. "${rscripts}"/functions
+
+start_test
+
+backup_log=$tmp/cancel-backup-checkpoints.out
+restore_log=$tmp/cancel-restore-checkpoints.out
+
+rm -f $backup_log
+rm -f $restore_log
+
+slowjob="slow-backup-bareos-fd"
+
+cat <<END_OF_DATA >"$tmp/bconcmds"
+@$out /dev/null
+messages
+@$out $backup_log
+run job=$slowjob level=Full yes
+quit
+END_OF_DATA
+
+run_bconsole
+
+sleep 14
```

Comment:
Waiting a fixed time is usually a bad idea. Better use a while loop with timeout.